docs/reference/run.md

```diff
@@ -293,6 +293,10 @@ By default, the MAC address is generated using the IP address allocated to the
 container. You can set the container's MAC address explicitly by providing a
 MAC address via the `--mac-address` parameter (format:`12:34:56:78:9a:bc`).
 
+Be aware that Docker does not check if manually specified MAC addresses are
+unique. This is by design, because a container can be connected to a different
+L2 domain.
```

Comment:
I would end after `unique.` I think that is the only info needed by the user who wants to play with setting the container's MAC address. Reason why docker does not monitor uniqueness of user chosen mac address may be several, from technical to philosophical or simply practical and not relevant to the user guide context, in my opinion.